### ISSUE TITLE
Validation silently returns incorrect result on second validation

### ISSUE THREAD

**Issue by joep** [trailblazer/reform#447]
### Complete Description of Issue

Validation silently fails (i.e., returns incorrect result) when a form is validated twice against the same instance. 

### Steps to reproduce

```
require 'reform'
require 'dry-validation'
require 'reform/form/dry'
require 'pp'

Reform::Form.class_eval do
  include Reform::Form::Dry
end

class AlbumForm < Reform::Form
  property :title
  validation do
    required(:title).filled(:str?)
  end
end

@form = AlbumForm.new(Struct.new(:title).new)
pp @form.validate({title: 'test'})  # ==> true

@anotherform = AlbumForm.new(Struct.new(:title).new)
pp @anotherform.validate({})  # ==> false
pp @anotherform.validate({title: 'test'})  # ==> also return false!

```

### Expected behavior

Final validation should return true.

### Actual behavior

Final validation returns false.

### System configuration

2.3.0.rc1

**Comment by apotonick** [comment]
The second call to `validate` is not how it's supposed to be used.

**Comment by joep** [comment]
Hmmm, I guess that just seems non-intuitive to me and works a bit differently than other validation systems such as activemodel.

Silently giving the wrong answer with valid input parameters just seems strange to me.

**Comment by apotonick** [comment]
Yes, it is non-intuitive as Reform's API is _wrong_ on this and shouldn't allow separate `validate` calls. This will be changed in Reform 3.

**Comment by apotonick** [comment]
FYI, I wrote about this issue on my blog a few years ago. https://apotonick.wordpress.com/2014/05/22/rails-misapprehensions-single-responsibility-principle/

**Comment by fran-worley** [comment]
@apotonick

To avoid confusion couldn't we just raise an error (you can't call validate on a validated contract!) ??

On 17 Sep 2017, at 09:01, Nick Sutterer <notifications@github.com<mailto:notifications@github.com>> wrote:


FYI, I wrote about this issue on my blog a few years ago. https://apotonick.wordpress.com/2014/05/22/rails-misapprehensions-single-responsibility-principle/

**Comment by apotonick** [comment]
Hm, I always hated the `DoubleRenderError` in Rails :grimacing: I rather change the API so this becomes impossible.

**Comment by fran-worley** [comment]
Oh I agree but surely in the meantime a double render error is better than a silent fail?

**Comment by emaglio** [comment]
This will be fix with v3.0
